chain/db/stateGenerator.go

```diff
@@ -73,6 +73,23 @@ func (cs *ChainStore) spendTransaction(states *StateDB, txn *transaction.Transac
 		}
 		states.UpdateBalance(donationAddress, Fixed64(genID.RegistrationFee), Addition)
 
+	case pb.NanoPayType:
+		nanoPay := pl.(*pb.NanoPay)
+		pg, err := txn.GetProgramHashes()
+		if err != nil {
+			return err
+		}
+
+		addrRecipient := BytesToUint160(nanoPay.Recipient)
+		nanoPayBalance, _ := states.GetNanoPay(pg[0], addrRecipient, nanoPay.Nonce)
```

Comment:
I think GetNanoPay should also return an error in addition, otherwise if there is an error, returned balance will be zero, and the claimAmount will be the latest amount

Reply:
done